src/wp-includes/class-wp-block-templates-registry.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Templates registry functions.
+ *
+ * @package WordPress
+ * @since 6.7.0
+ */
+
+/**
+ * Core class used for interacting with templates.
+ *
+ * @since 6.7.0
+ */
+final class WP_Block_Templates_Registry {
+	/**
+	 * Registered templates, as `$name => $instance` pairs.
+	 *
+	 * @since 6.7.0
+	 * @var WP_Block_Template[] $registered_block_templates Registered templates.
+	 */
+	private $registered_templates = array();
+
+	/**
+	 * Container for the main instance of the class.
+	 *
+	 * @since 6.7.0
+	 * @var WP_Block_Templates_Registry|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Registers a template.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param string $template_name Template name including namespace.
+	 * @param array  $args          Optional. Array of template arguments.
+	 * @return WP_Block_Template|false The registered template on success, or false on failure.
```

Comment:
This doesn't match the actual return type. This method may also return a `WP_Error` object.